Foodening

MTandi opened the pull request:
## About The Pull Request

Adds stronger mood buffs acquired by consuming complex meals with natural ingredients.

Design doc: https://hackmd.io/@ZSLHD7nTTlWNegwCVVdJBw/ryqRgDluj

## Why It's Good For The Game

More depth to the chef and botanist professions. 
Rewarding effort spent on acquiring good ingredients and making complex food.

## Changelog

:cl:
add: Mood buff power and duration depends on the ingredient quality and recipe complexity of the food.
add: Hand-crafted food gives other buff. Buff type depends on recipe complexity and duration - on ingredient quality.
code: Added infrastructure to assign custom buff to a certain dishes for unique effects
add: Consumable reagents now care about purity, being more nutritious with higher purity. Dispensers give synthetic food, natural sources are organic and better and give longer buffs.
add: Grown food reagent purity now depends on lifespan, endurance and instability stats of the seed. 
add: Natural booze now uses the same purity formula and 100% purity can increase booze power by up to 200%.
add: Added Fat reagent that is now a base type for lard, vegetable oil, olive oil. Corn oil and cooking oil merged into vegetable oil.
balance: Reagents and their purities are preserved during food chemical reactions, processing, cooking and crafting.
balance: Sugar nutrition rebalanced due to it being 100% pure in chem dispenser, overdose happens at 100u now.
fix: `grind_results` of grown food give full amount equal to all removed nutrients instead of just `/datum/reagent/consumable/nutriment`
qol: Seed Extractor now shows additional seed data: possible mutations, juicing result, grinding result, fermentation result
fix: Reverts the unintended change of monkey cube biomass cost back to 50
/:cl:

MTandi committed:
grown reagents formula

MTandi committed:
purity for mapspawn and biogen stuff

MTandi committed:
seed purity generalized, natural booze changes

MTandi committed:
nutriment factor scaled with purity

MTandi committed:
starting reagent purity and meats

MTandi committed:
organic milk and eggs

MTandi committed:
egg-plant egg purity

MTandi committed:
meatwheat purity

MTandi committed:
mood events and examine list

MTandi committed:
reagents

ChungusGamer666 commented:
lovely pr but its gonna conflict with my tongue pr so bad
its so over

MTandi commented:
> lovely pr but its gonna conflict with my tongue pr so bad its so over

I'll handle it, it's fine. It will conflict with many other things before I complete it.

Loafin34 commented:
i feel like this is a lot of work for not a lot in general. will you fix the produce order stuff? like, pine nuts, moon jellies, san marzano tomatoes, that stuff is only available from your own wallet.

AtomTheProphet commented:
Maybe better manipulators/servos for the biogenerator could also increase the reagent purity, rather than it always being 0.5? Like, it could maybe be 0.75-0.9 with T4 parts?

the-orange-cow commented:
For consideration, maybe replace endurance with potency for food purity?  Ultimately your call, but potency seems like it would be more intuitive.  As far as the lifespan aspect, I would assume this is a timegate?

CRITAWAKETS commented on code/modules/research/designs/biogenerator_designs.dm:
```diff
@@ -3,67 +3,67 @@
 
 /datum/design/enzyme
-	name = "Universal Enzyme"
+	name = "Synthetic Universal Enzyme"
```
Synthetic universal enzyme seems a little weird, considering how universal enzyme in itself is already likely synthetic, plus considering how it's not consumed in recipes this'll likely just make this recipe completely useless.

MTandi replied:
It is a protein, it does fill you up (although just 1 nutriment factor), you can make a pure version of it by fermenting tomatoes and the purity of catalysts will affect the purity of the result product. Since 100% pure milk is easily obtained from goat, you would make finest cheese roundstart if this wasn't considered. I may rename it to "Synthetic Enzyme" though.

CRITAWAKETS replied:
Might wanna change the bottle of enzyme in the chef's kitchen to this, then.

MTandi replied:
Mapspawn things are better than biogen stuff on purpose.

CRITAWAKETS commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -94,11 +94,14 @@
 	for(var/datum/surgery/surgery as anything in exposed_carbon.surgeries)
 		surgery.speed_modifier = max(0.1 * power_multiplier, surgery.speed_modifier)
 
+/datum/reagent/consumable/ethanol/synthetic
```
renember to edit chem dispenser stuff and recipes

CRITAWAKETS commented on code/modules/reagents/chemistry/reagents/food_reagents.dm:
```diff
@@ -239,13 +243,16 @@
 	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
 	default_container = /obj/item/reagent_containers/condiment/sugar
 
+/datum/reagent/consumable/sugar/synthetic
```
Renember to edit the chem dispenser to dispense this.

CRITAWAKETS replied:
True on the part about affecting chemist gameplay, That said, I foresee this as probably just bringing back the old strat of botany always making a chem dispenser, though the chef might also do it. Plus, the bartender benefits heavily from it and would benefit even more.

Actually now that I realize it, the soda dispenser makes sugar too, that's... already in service.

MTandi replied:
Yeah, I have in my TODO to check whether any other consumables affect chem things. May adjust things for sugar and other things if any.

bluetshirtguy commented:
Neat pr but food is a quite the bother as is, more micro managing doesn't seem particularly nice for cooks.

MTandi commented:
> Maybe better manipulators/servos for the biogenerator could also increase the reagent purity, rather than it always being 0.5? Like, maybe it could be 0.75-0.9 with T4 parts?

That would negate the point of natural ingredients. Biogen food is enough to fill yourself, but it doesn't bring additional bonuses.

> For consideration, maybe replace endurance with potency for food purity? Ultimately your call, but potency seems like it would be more intuitive. As far as the lifespan aspect, I would assume this is a timegate?
> 
> Edit: To clarify why I think potency would be more intuitive than endurance, the size of the plant increases with potency, giving the chef an easy visual way of identifying which produce is likely to be of better quality as an ingredient.

Potency is the default stat everyone max out regardless, so they will be making the best ingredients by accident. And potency already has an important effect - increasing the volume of reagents. Lifespan and Endurance, on the other hand, are usually ignored by botanists because they don't affect your harvest in any way. And they are tied to health pool, so it makes perfect sense - Potency makes your apple big, but weak. Lifespan and Endurance make it strong, and have pure reagents. I thought of also tying these stats to decomposition speed later.

Also it leads to another gameplay branch - people will explore grafting and pollution more to get the best ingredients. They will be perfectly fine with mediocore reagents if they don't want to bother. 0.5 purity leads to the same results as right now and 1 purity will just double the mood buff duration.

MTandi commented:
> Neat pr but food is a quite the bother as is, more micro managing doesn't seem particularly nice for cooks. edit: not to mention tedium and specifics to demand of botanists who are notoriously inconsistent.

Don't bother then. Cook with mediocore reagents.

AtomTheProphet commented:
> > Maybe better manipulators/servos for the biogenerator could also increase the reagent purity, rather than it always being 0.5? Like, maybe it could be 0.75-0.9 with T4 parts?
> 
> That would negate the point of natural ingredients. Biogen food is enough to fill yourself, but it doesn't bring additional bonuses.

Why would "natural" ingredients be better anyway? I'd assume that being able to control every single component that into something like milk would make its quality quite high and consistent, and I'm pretty sure with better parts you're only able to more effectively control what goes into it. 

I don't know, it just sounds like you're nerfing the biogenerator just because you want people to invest more time into botany at the cost of making being a decent cook harder, more time consuming, and tedious even when it doesn't make any form of sense. That's not even mentioning the fact that you can't always rely on botanists to grow stuff for you to cook with.

MTandi commented:
> I don't know, it just sounds like you're nerfing the biogenerator just because you want people to invest more time into botany at the cost of making being a decent cook harder, more time consuming, and tedious even when it doesn't make any form of sense. That's not even mentioning the fact that you can't always rely on botanists to grow stuff for you to cook with.

There is NO nerf, read the doc. You didn't get mood buff unless the food was in your species diet before. Now the buffs are stronger and can be obtained even when the food doesn't match your diet. Diet only adds to the strength of the buff. And purity only affects the duration of the buff, so you will be perfectly fine with biogen ingredients and you get the mood buffs regardless.

Synthetic prefix in biogen is needed to hint that this mechanic exists.

MTandi commented:
> I just realized you were talking about lifespan, and not plant age. To my knowledge, lifespan is hardcoded into each individual plant and isn't affected by any fertilizer. It can't really be adjusted without just using a separate base plant. Not sure if this is pertinent to how you want the mechanic to work, but it bears mentioning as there really isn't much room for choice with that stat.

You can change it by grafting weeds with high lifespan onto the plant. Also I think instability may adjust it at random? Not sure. I may add an effect of boosting lifespan to some reagent later to provide an alternative option.

> Also, if you don't mind me asking, how would the instability randomization portion of the mechanics work? Would all of a harvest be uniform with whether they get the plus or minus bonus to purity, or would it be individual to each individual piece of produce?

Its random from -0.25 purity to +0.25 purity at 100 instability. Lower instability - less spread. You get uniform harvest only at 0 instability. It is individual to every piece of produce within the harvest.

> Also also(and I apologize if this is obvious, I don't know much about fermi/purity code), is this food purity the same as other chemical purity? As in, if these mechanics were utilized to grow medicine instead of foodstuffs, could pure medicine be obtainable via botany?

It's the same. Yes, pure medicine can be obtained via botany, but it's quite tedious due to lifespan and endurance requiring time to max out.

bluetshirtguy commented:
> > Neat pr but food is a quite the bother as is, more micro managing doesn't seem particularly nice for cooks. edit: not to mention tedium and specifics to demand of botanists who are notoriously inconsistent.
> 
> Don't bother then. Cook with mediocore reagents.

So you mean spend more time yourself in the brief 40 minutes you have OR create higher expectations(less consistent) and if you don't do either you're now mediocre? ouch.

MTandi commented:
> So you mean spend more time yourself in the brief 40 minutes you have OR create higher expectations(less consistent) and if you don't do either you're now mediocre? ouch.

Just don't read this PR and you won't notice anything as a chef except that your food now has better buffs than before.

You can explore chemistry without the need to bother a lot with PH and buffers.
You can do atmos without doing crystalizer.

It's the same. This is a geeky mechanic and you won't be punished if you don't engage with it. Just an additional thing to do if you are bored.

SomeoneYouProbablyKnow commented:
I wish we had a design doc for mood as a whole (maybe we do but i couldnt find it) because this PR looks good and i want to complain about mood.

MTandi commented:
> I wish we had a design doc for mood as a whole (maybe we do but i couldnt find it) because this PR looks good and i want to complain about mood.

It's WIP, I will provide more details when I figure out the food complexity. Like, pastry base requires a lot of steps, so donuts should be an exceptionally complex food? Probably not, since they're around the station roundstart. The gist is stated in the OP message - Recipe complexity and matching diet give stronger mood buff, purity (synthetic vs organic) affect the duration of the buff and how filling the food is.

Profakos commented:
As someone who plays chef a lot, this is very interesting to me.

But also: as long as you are touching Natural Booze, could you please look into the issue that renaming a reagent breaks using them in the chemmaster, due to chemicals being ID-d by their name? Same issue persists for fruit wines, so I couldn't make Berry Wine Flavour Packets. Thank you!

Likteer commented:
To all them haters - remember that making plants healthier, is as easy as burning paper and scooping up the resulting ash for fertiliser.

MTandi commented:
> But also: as long as you are touching Natural Booze, could you please look into the issue that renaming a reagent breaks using them in the chemmaster, due to chemicals being ID-d by their name? Same issue persists for fruit wines, so I couldn't make Berry Wine Flavour Packets. Thank you!

Already fixed via #75849 

>To all them haters - remember that making plants healthier, is as easy as burning paper and scooping up the resulting ash for fertiliser.

It's lifespan (stat responsible for max health pool) and not the current health of plant.

Profakos commented:
> 



> > But also: as long as you are touching Natural Booze, could you please look into the issue that renaming a reagent breaks using them in the chemmaster, due to chemicals being ID-d by their name? Same issue persists for fruit wines, so I couldn't make Berry Wine Flavour Packets. Thank you!
> 
> Already fixed via #75849
> 

Aha, sorry, didn't know about that! Thank you!

AtomTheProphet commented:
> > I don't know, it just sounds like you're nerfing the biogenerator just because you want people to invest more time into botany at the cost of making being a decent cook harder, more time consuming, and tedious even when it doesn't make any form of sense. That's not even mentioning the fact that you can't always rely on botanists to grow stuff for you to cook with.
> 
> There is NO nerf, read the doc. You didn't get mood buff unless the food was in your species diet before. Now the buffs are stronger and can be obtained even when the food doesn't match your diet. Diet only adds to the strength of the buff. And purity only affects the duration of the buff, so you will be perfectly fine with biogen ingredients and you get the mood buffs regardless.
> 
> Synthetic prefix in biogen is needed to hint that this mechanic exists.

I may be misunderstanding something. Is the nutritional value of the food made in the biogen the same as it was before? I'm fine with giving bonuses to food made with botany-grown ingredients, but I don't think it's a good idea to punish cooks for using the biogen.

MTandi commented:
> I may be misunderstanding something. Is the nutritional value of the food made in the biogen the same as it was before? I'm fine with giving bonuses to food made with botany-grown ingredients, but I don't think it's a good idea to punish cooks for using the biogen.

The nutrition is a bit lower yes (if you don't purify it in chromatography machine), but it is balanced out by making simplest recipes give mood buff (walk and action speed increase). If you use chromatography machine, then the nutritional value is the same as before.

AtomTheProphet commented:
> > I may be misunderstanding something. Is the nutritional value of the food made in the biogen the same as it was before? I'm fine with giving bonuses to food made with botany-grown ingredients, but I don't think it's a good idea to punish cooks for using the biogen.
> 
> The nutrition is a bit lower yes (if you don't purify it in chromatography machine), but it is balanced out by making simplest recipes give mood buff (walk and action speed increase). If you use chromatography machine, then the nutritional value is the same as before.

So you're punishing people for using the biogen. I don't think that's the best idea mate. Even if the simplest recipes do give you a mood buff, you'll still need to actually eat more of the food to actually stop being hungry.

CRITAWAKETS commented on code/game/objects/items/food/_food.dm:
```diff
@@ -64,6 +66,8 @@
 	make_decompose(mapload)
 	make_bakeable()
 	make_microwaveable()
+	if(!starting_reagent_purity)
```
should be if(starting_reagent_purity), this just automatically returns when it gets to the proc.

MTandi commented:
> So you're punishing people for using the biogen. I don't think that's the best idea mate. Even if the simplest recipes do give you a mood buff, you'll still need to actually eat more of the food to actually stop being hungry.

I will consider bumping the biogen stuff to 0.4 purity from 0.3 so that the nutritional decrease will be only 20% instead of 40%.
There will be more serious nutrition changes in this PR later as right now some recipes made nutrition out of nowhere. I may make the nutrition scale from 0 to 300% with purity then biogen stuff will be the same as before in terms of nutrition.

CRITAWAKETS commented on code/modules/hydroponics/seeds.dm:
```diff
@@ -302,6 +303,14 @@
+	var/purity_from_lifespan = lifespan / 400 //up to +25% for lifespan
+	var/purity_from_endurance = endurance / 400 //up to +25% for endurance
+	var/purity_from_instability = rand(-instability, instability) / 400  //up to +-25% at random for instability
+	var/result_purity = clamp(0.5 + purity_from_lifespan + purity_from_endurance + purity_from_instability, 0, 1) //50% base + stats
```
Should probably either increase the bonuses or add a new factor, since even with maximum lifespan and endurance, if there's _any_ instability you've got a 50% chance of your produce not being at 100% purity which is kind of disappointing.

MTandi replied:
it will be only +-2.5% of purity if you have 10 instability. You can easily remove it with robust harvest or diethylamine. And purity is an invisible stat (you can only figure out the exact value in chromatography machine) so reaching 100% purity is not the goal. It may even be possible for admin spawn stuff only, or if you know how to cross-pollinate stuff onto plants with 100 lifespan.

CRITAWAKETS replied:
You can actually find purity with the Chemical analyzer, which is handheld.

MTandi replied:
Well anyway, 98% pure reagents will mean that you get the 10 minute buff reduced by like 12 seconds. And they will be only 2% less filling, which is negligible. The effects are scaled, so 100 is just a cap, rather than a goal.

CRITAWAKETS commented:
So, just going to note that Soda Water, Tonic Water and Ice are all marked as consumable. Might wanna give those 100% purity by default, unless you want bartenders to figure out how to freeze their own ice.

MTandi commented:
> So, just going to note that Soda Water, Tonic Water and Ice are all marked as consumable. Might wanna give those 100% purity by default, unless you want bartenders to figure out how to freeze their own ice.
> 
> Would also be neat if bottled goods were above 50% purity, so a bartender wishing to make an "expensive" drink without special ingredients from botany could use the booze-o-mat.

I will take a closer look at bar in later PR, for now the goal is just to make it unaffected if possible.

MTandi committed:
mood effects into globals

MTandi committed:
Merge branch 'master' into food_reagents

MTandi committed:
fixed merger issues

MTandi committed:
synthetic enzyme

MrMelbert commented on code/datums/components/food/edible.dm:
```diff
@@ -214,6 +214,30 @@ Behavior that's still missing from this component that original food items had t
 		var/list/types = bitfield_to_list(foodtypes, FOOD_FLAGS)
 		examine_list += span_notice("It is [lowertext(english_list(types))].")
 
+	if(istype(parent, /obj/item/food))
+		var/obj/item/food/food = parent
+		var/quality = get_preceived_food_quality(user, food)
+		if(quality > 0)
+			var/quality_label = GLOB.food_quality_description[quality]
+			examine_list += span_green("You find this meal [quality_label].")
+		else if (quality == 0)
+			examine_list += span_notice("You find this meal edible.")
+		else
+			examine_list += span_warning("You find this meal inedible.")
+		var/purity = food.reagents.get_average_purity()
+		switch(purity)
+			if(0 to 0.2)
+				examine_list += span_warning("The ingredients are terrible.")
+			if(0.2 to 0.4)
+				examine_list += span_warning("The ingredients are synthetic.")
+			if(0.4 to 0.6)
+				examine_list += span_notice("The ingredients are mediocore.")
+			if(0.6 to 0.8)
+				examine_list += span_green("The ingredients are organic.")
+			if(0.8 to 1)
+				examine_list += span_green("The ingredients are finest.")
```
Food quality should not be locked to `/obj/item/food`, it is a template type intended solely for easy application of the edible component.

MrMelbert commented on code/datums/components/food/edible.dm:
```diff
@@ -498,26 +522,58 @@ Behavior that's still missing from this component that original food items had t
+		H.add_mood_event("toxic_food", /datum/mood_event/disgusting_food)
+		return
+
+	if(!istype(parent, /obj/item/food))
+		return
```
Same here

MTandi replied:
This one checks for `venue_value` so it has to remain.

MrMelbert commented on code/modules/hydroponics/grown.dm:
```diff
@@ -111,30 +111,31 @@
+	var/quality = round(LERP(quality_min, quality_max, purity_above_base))
 	for(var/datum/reagent/reagent in reagents.reagent_list)
-		if(reagent.type != /datum/reagent/consumable/nutriment && reagent.type != /datum/reagent/consumable/nutriment/vitamin)
+		if(!istype(reagent, /datum/reagent/consumable))
```
Is there a reason why this is changing?

MTandi replied:
I want to make biogen work with any consumable, respecting nutrition values, instead of just converting 1 nutriment or 1 vitamin to 1 biomass. Not in code yet.

MrMelbert replied:
This'll cause a lot more than sugar to be fermented, like other alcohol, so I'm not sure it's super what we want. Maybe it should be a typecache?

MTandi replied:
Well this is a ferment() proc that is called when you put fruit into a barrel or it is fermented in the barrel-melon. I believe it is it's fine that other alcohol reagents will be turned into the other. I actually can't think of real case when this should happen unless you inject fruit with some other alcohol on purpose.

MrMelbert reviewed (changes requested):
Generally speaking, if you're doing `istype(x, /obj/item/food)`, it is an error (see below)

ATH1909 commented:
how does this interact with chem recipes that require sugar as a component? e.g. mannitol?

with moth clothes eating?

aren't monkeys how chefs are supposed to get their meat?

MTandi commented:
> how does this interact with chem recipes that require sugar as a component? e.g. mannitol?

TBD. The goal is to make chem unaffected. Maybe tweaking sugar nutriment factor to not buff sugar pills by accident.

> with moth clothes eating?

No planned changes in scope of this PR, but I may take a look at those and tie it somehow to the purity of cellulose fibers.

> aren't monkeys how chefs are supposed to get their meat?

You can breed cows, chickens, grow meat wheat, order meat crates. Biogen will become the less rewarding way of aquiring ingredients since it's essentially effortless.

MTandi committed:
logic into edible

MTandi committed:
food quality is universal

MTandi committed:
unit test fix

MTandi committed:
linting

MTandi committed:
recipe complexity base logic

AtomTheProphet commented:
> > So you're punishing people for using the biogen. I don't think that's the best idea mate. Even if the simplest recipes do give you a mood buff, you'll still need to actually eat more of the food to actually stop being hungry.
> 
> I will consider bumping the biogen stuff to 0.4 purity from 0.3 so that the nutritional decrease will be only 20% instead of 40%. There will be more serious nutrition changes in this PR later as right now some recipes make nutrition out of nowhere. I may make the nutrition scale from 0 to 300% with purity then biogen stuff will be the same as before in terms of nutrition.

It would still be greatly appreciated if you were to do this.

MTandi committed:
some food complexities

MTandi committed:
conflicts resolved

MTandi committed:
rest of food complexities

MTandi committed:
fix nutriments

Loafin34 commented:
could you also fix bartender ice cups? they start at 300 kelvin (or something along those lines) which makes ice cups instantly melt. that is not a good thing.

SyncIt21 commented:
Love the PR great work. I have a suggestion and you are free to ignore it i don't mind. So this is what i had in mind

there was this comment https://github.com/tgstation/tgstation/pull/76317#issuecomment-1609959805 and it makes sense so here is what i thought. Rather than decreasing the nutrition value of flour & sugar created by the bio generator why not make the bio-gen extract Flour & Sugar from Wheat & Sugarcane directly? That way they can keep the natural nutrition value while also giving Wheat & Sugar cane more importance so they can be grown more

You could have 2 separate progress bars in the bio-gen UI say Flour & Sugar that only increases when the bio-gen processes Wheat & Sugarcane and it uses these values when dispensing them. So when you grind up wheat & sugarcane you also get nutrients that could be added as biomass back into the biogen. Better parts means it better extracts Flour & Sugar from the wheat & sugarcane it starts at 0.5 and then becomes 1(meaning 100% extraction) the unconverted amount with inefficient parts will be added back as bio-matter.

Again just an idea i had, but idk what are your thoughts on it?

MTandi commented:
@SyncIt21 This PR is already going to be quite heavy, with soup changes and everything else. I don't plan to touch biogen other than making its products slightly worse than default/grown ones.

SyncIt21 commented:
> @SyncIt21 This PR is already going to be quite heavy, with soup changes and everything else. I don't plan to touch biogen other than making its products slightly worse than default/grown ones.

Fair enough but would you consider it in an future PR after this get's merged or do you have any arguments against it?

MTandi commented:
> Fair enough but would you consider it in an future PR after this get's merged or do you have any arguments against it?

Biogen's purpose is a kind of a service protolathe that works off nutriments instead of materials. It would not be correct to turn it into a sort of grinder. And I don't like to overcomplicate things to make biogen look like hypertorus UI. Service jobs are usually a choice of new players, so unnecessary complexity may harm their onboarding with the game.

I think biogen is fine as it is right now, and nerfing its nutriments in this PR is enough because it's essentially free and most lazy source of milk/flour/sugar. 

If you want to adjust its logic, then feel free to create a PR, maybe this is a good idea. I'm just not convinced.

SyncIt21 commented:
True that makes sense. It's just that Wheat and sugarcane have no significance because of the biogen so I thought this would be a good idea but yeah I would be changing the purpose of it. 

Fair enough thanks for answering

MTandi committed:
Merge branch 'master' into food_reagents

MTandi committed:
fats and oils

MTandi committed:
Marblle meat

MTandi committed:
food reaction purity

MTandi committed:
food buffs

MTandi committed:
weighted buffs

MTandi committed:
reagents are fully retained during crafting

MTandi committed:
processor doesnt create new reagents

MTandi committed:
microvave doesn't create new reagents

MTandi committed:
grilling doesn't create new reagents

MTandi committed:
baking doesn't create reagents

MTandi committed:
reagentgrinder mixing

MTandi committed:
juicing refactored

MTandi committed:
butter balance

MTandi committed:
holder helpers

MTandi committed:
typo

MTandi committed:
juice into food

MTandi committed:
consistent juicing

MTandi committed:
fuck this

MTandi committed:
drying preserves purity

MTandi committed:
wording

MTandi committed:
seed extractor labels

MTandi committed:
sugar issue

MTandi committed:
monkey cubes price fixed

MTandi committed:
Hyperglycaemic shock at 100u sugar

MTandi committed:
Merge branch 'master' into food_reagents

MTandi committed:
oops

MTandi committed:
removing debug stuff

MTandi committed:
typo

MTandi committed:
added_purity

MTandi committed:
nutrition estimation removed from crafting

jlsnow301 commented on code/modules/food_and_drinks/recipes/tablecraft/recipes_cake.dm:
```diff
@@ -142,7 +142,7 @@
 	category = CAT_CAKE
 
 /datum/crafting_recipe/food/bscccake
```
can you change this now that its no longer accurate (its also unreadable)

jlsnow301 commented on tgui/packages/tgui/interfaces/PersonalCrafting.tsx:
```diff
@@ -832,14 +832,14 @@ const RecipeContent = ({ item, craftable, busy, mode, diet }, context) => {
                 />
               )}
-              {item.nutriments > 0 && (
+              {item.complexity !== undefined && (
```
```suggestion
              {!!item.complexity && (
```

MTandi replied:
This would not allow complexity when it is 0.

MTandi committed:
cake names

ExcessiveUseOfCobblestone commented:
Is there not a concern here for being pigeonholed into making specific foods based on what buffs they provide in addition to "powergaming" the chef role?

Making the complex meal because its complex vs. making the complex meal because it gives you combat buffs.

Do we really want people to rush the same food every round because it is the most efficient to make to get the most optimal buffs?

This gets to especially be a problem when people want to not eat lower tiered food because they dont want to get full so they can maximize the buffs from the stronger tier. 

This also bleeds into botany where now they are sandbagging the chef if they dont make the plants used for the meta strat, although tbf thats not a new problem its just more specialized.

The combat buff aspect is something im pretty weary of, I think Id rather have quality of products use be limited to the amount of servings provided by the meal (personally complex meals should have buffet-levels of servings considering the amount of ingredients needed) and the duration of the mood bonuses.

The importance of food should never be so much that the chef has to forego the enjoyment of say making a specific themed restaurant because of the fear of inefficiency and the social ridicule/conflict that arises from that.

ExcessiveUseOfCobblestone reviewed (changes requested):
DNM while stun immunity and noslip are in.

Shock resistance is "ok", immunity is almost DNM for me.

Just putting these here in case another maint is interested in the PR, I am still mulling over the concept in general.

MTandi committed:
unit test fix

MTandi commented:
> Is there not a concern here for being pigeonholed into making specific foods based on what buffs they provide in addition to "powergaming" the chef role?

- The strong buffs are given for only the most complex food, which is barely created in a shift because of required effort to gather ingredients. So it's a pain to achieve it first. 
- Then there is a random roll between multiple buffs, so your desired buff is not guaranteed. 
- And you can only get one buff from one meal - only unbitten food gives it. You have a room for error if you make a pizza and cut it though.

I slapped those buffs just for the sake of adding something for now, planning to add some more flavorful non-random buffs to different food. Like eating pasta would make you speak Italian briefly. Clown food makes you wobble walk. Etc. 
There is a system for easy addition of such buffs and I hope the community will come up with their own ideas for different dishes later.

I'm fine removing all trait buffs for now, leaving only the basic move speed buff that scales with complexity.

> Making the complex meal because its complex vs. making the complex meal because it gives you combat buffs.
> 
> Do we really want people to rush the same food every round because it is the most efficient to make to get the most optimal buffs?

It's random, unless certain food has a particular buff specified. And high complexity food are painfully hard to make. And again, the buff don't have to be combat related.

> This gets to especially be a problem when people want to not eat lower tiered food because they dont want to get full so they can maximize the buffs from the stronger tier.

You get the buff with the first bite, you don't need to eat the entire meal to get the buff.

> This also bleeds into botany where now they are sandbagging the chef if they dont make the plants used for the meta strat, although tbf thats not a new problem its just more specialized.

Plant stats only affect the duration of the buff.

> The importance of food should never be so much that the chef has to forego the enjoyment of say making a specific themed restaurant because of the fear of inefficiency and the social ridicule/conflict that arises from that.

Right now the buffs are not tied to food types or categories, they are completely random, and every category has complexity 4 or 5 foods, so they can keep up with the theme.

MTandi committed:
removed noslip and stunimmune buffs from the pool

MTandi committed:
revert effect logic change

MTandi committed:
typo

MTandi committed:
placeholder icons for hud statuses

EOBGames commented:
Just to mirror my thoughts from a discord discussion with the author:

I don't like this PR, because I feel that chef is first and foremost a creativity-based job wherein the freedom to make what they want without locks or meta considerations is key to the experience. While I do ultimately agree that the chef currently suffers from an issue of lowest-common-denominator cooking (easy food is easy, people like easy shifts), I don't believe the solution to that lies in creating a buff-enforced meta hierarchy of foods which pigeonholes the chef away from making what they want to make. I also sympathise with the intention to make chef more relevant to a given round, but it's the current lack of rigid meta choices and, to an extent, their less-than-necessary nature in a round that gives chefs the opportunity to explore food options and be creative with their choices. Anything that's going to lead to, for example, powergamers coming up to the desk and requesting specific food, is imo the exact wrong direction for chef to take.

Additionally I need to talk about the proposed TODOs in the hackmd- the proposed locking system in particular. While the above stuff only functionally creates a meta-enforced limitation on creativity, this proposal would entirely codify it. The strength of chef as an experience relies upon being able to do what you want. I feel it's nothing but detrimental to the job to take this away and force chef players to make a bunch of stuff they don't want to before they can even begin to get started on actually enjoying their shift. Species-locked recipes is an even worse thing to stack on top of this, and goes against a specific design decision on species food- which was it should be available so that people can actually make them and get them seen in a round, partially because they're neat food options and partially as a readily-available link to the server lore.

MTandi commented:
> Anything that's going to lead to, for example, powergamers coming up to the desk and requesting specific food

I believe it is on the opposite, a good thing. People come to chemistry for specific reagents, and I don't see chemist being bothered with this undermining their "creative job". 

Chef has CQC in case if powergamers will be annoying.

>Additionally I need to talk about the proposed TODOs in the hackmd

These changes are unrelated with this PR and may never happen.

MTandi committed:
Update tgui/packages/tgui/interfaces/PersonalCrafting.tsx

Co-authored-by: Jeremiah <42397676+jlsnow301@users.noreply.github.com>

EOBGames commented:
> I believe it is on the opposite, a good thing. People come to chemistry for specific reagents, and I don't see chemist being bothered with this undermining their "creative job".

The difference between chemist and chef is that one of the jobs is currently given full creative freedom and the other isn't. The existence of chemist as a non-creative job doesn't translate to a reason to reduce the creative element of chef. Also- the reason that chemists aren't presently bothered by their general lack of creativity is because of the chem factory, MDs being given pharmacy access to produce their own medication, and botany's general reliance on unstable mutagen being reduced (i.e. chemists were made less important to the running of the round)- prior to those changes, when chemist was the primary source of chems on the station, I very much remember them getting annoyed by the constant requests distracting them from what they wanted to do with their shift.

> And bots already do exactly this, they request specific food in exchange for money. As a chef you can ask these money from players or make them get ingredients for what they want.

Bots are entirely, 100% optional and nobody is going to be upset at chef players if they don't engage with them. I don't view their existence as a reason to reduce the freedom chefs have in how they serve actual players in the round.
> Chef also has CQC in case if powergamers will be annoying.

Tying the ability of chef players to enjoy a shift to being able to fight the people annoyed at them isn't a solution to the problem.

Wallemations commented:
Can you either add any major sprite changes to the PR or just comment a screenshot of the spritediff bot output, I'm on mobile and will continue to be so for the rest of tomorrow so

MTandi commented:
> Can you either add any major sprite changes to the PR or just comment a screenshot of the spritediff bot output, I'm on mobile and will continue to be so for the rest of tomorrow so

![image](https://github.com/tgstation/tgstation/assets/3625094/91c0a95b-6def-46c8-be84-7ad47f10d825)

Wallemations reviewed (approved):
Cool thanks

MTandi committed:
Merge branch 'master' into food_reagents

MTandi committed:
debug

MTandi committed:
reagent transfer

MTandi committed:
reduced buff list

MTandi committed:
Merge branch 'master' into food_reagents

MTandi committed:
cleanup

MTandi committed:
unintended

the-orange-cow commented:
A couple things based off the recent test merge.  

I believe something in this pull may break the botany "auto juicing" plant trait.  Oranges with the trait didn't convert their nutriment into orange juice.  It is possible this also affects the "auto distilling" plant trait, since they operate in a similar matter.

Replacing corn oil with vegetable oil also has the effect of removing certain reaction that use corn oil as an ingredient, such as spray tan.

MTandi committed:
juicing fix

MTandi commented:
> I believe something in this pull may break the botany "auto juicing" plant trait. Oranges with the trait didn't convert their nutriment into orange juice. It is possible this also affects the "auto distilling" plant trait, since they operate in a similar matter.

Thank you, fixed the juicing, I removed a proc by accident. Distilling worked normally.

> Replacing corn oil with vegetable oil also has the effect of removing certain reaction that use corn oil as an ingredient, such as spray tan.

I tried spray tan and candy corn. Both work properly. What reactions did you have issues with?

the-orange-cow commented:
> I tried spray tan and candy corn. Both work properly. What reactions did you have issues with?

Disregard, I just realised something, made a mistake.  Sorry for false report.

MTandi commented:
I just realized that griding resets purity of grown food reagents, going to fix that too.

githubuser4141 commented:
Why are 'synethic' and 'organic' ingredients arbitairly terrible and excellent, that doesn't fit in with the whole 'set in the year 2500' part of the game; even today 'organic' doesn't necessairly equate to better quality food, in /TG/ it's already established that tech like: genemodding (in mammals), brain interfacing, faster then light travel, instant vaccines/medicine, medicine that actually *regenerates tissues*, and microfabrication (literally compressing 100's of irl supply lines into a printer-sized box, lolwut) exist. All that tech is wondertech that we'll likely never see IRL, when one can manufacture a *superior human heart* in a few seconds it's really bizarre that food science is still in the 21st century. It'd be fun to have a whole food chemistry system like what AtomTheProphet said, you'd get to precisely control what flavours/nutrients are present in your synthesized food.

MTandi committed:
grinding preserves purity

MTandi committed:
return fix

MTandi committed:
griding fix

MTandi committed:
unit test fix

MTandi committed:
more seed extractor data

MTandi commented:
Added some more data to Seed Extractor UI to reduce the need of wiki for botany.

![image](https://github.com/tgstation/tgstation/assets/3625094/230a4a6f-ea67-49fe-b98d-f50a285603a6)

MTandi committed:
deep fryer fix

MTandi commented:
Fixed #77286

MTandi committed:
wording

optimumtact commented:
have you removed the bits objected to yet?

MTandi commented:
> have you removed the bits objected to yet?

The list of available buffs got reduced to the default movespeed buff that scales with complexity, and there is a shock immunity buff at high level food as an example of buff that temporarily gives a trait.

Stun resist and anti slip got removed. 

Nothing else in my plans for this PR.

AtomTheProphet commented:
> Why are 'synethic' and 'organic' ingredients arbitrarily terrible and excellent, that doesn't fit in with the whole 'set in the year 2500' part of the game; even today 'organic' doesn't necessairly equate to better quality food, in /TG/ it's already established that tech like: genemodding (in mammals), brain interfacing, faster then light travel, instant vaccines/medicine, medicine that actually _regenerates tissues_, and microfabrication (literally compressing 100's of irl supply lines into a printer-sized box, lolwut) exist. All that tech is wondertech that we'll likely never see IRL, when one can manufacture a _superior human heart_ in a few seconds it's really bizarre that food science is still in the 21st century. It'd be fun to have a whole food chemistry system like what AtomTheProphet said, you'd get to precisely control what flavours/nutrients are present in your synthesized food.

Yeah. But most people don't care much about this stuff unless it's something like respriting the Mosin. This PR only creates an unnecessary and unrealistic distinction between food synthesized in a controlled environment compared to food grown in an environment that would fairly heavily influence the resulting product. 
Like, I'm fine with grown food have the potential to be better, but what I think is really unhealthy is that ingredients produced in a biogenerator are arbitrarily nerfed for no reason other than "It was made by a machine".

Having full control over the flavors and nutrients is a bit more complicated, but yeah.

carlarctg commented:
![image](https://github.com/tgstation/tgstation/assets/53100513/76eeb587-ec66-4066-96f9-c2c1ea3f6617)

This PR breaks grinding up mats for parts, making EMPs, lexorin, mute tox, etc, impossible to make. This likely affects other things like slime jelly too

MTandi committed:
grinding fix

MTandi commented:
> This PR breaks grinding up mats for parts, making EMPs, lexorin, mute tox, etc, impossible to make. This likely affects other things like slime jelly too

Fixed. It was not working for atoms that have no `/datum/reagents`. So it was working for slime jellies, fruits, food, but not for plasma sheets and other things that have no internal `reagents` by default.

rdragan commented:
Why does this increase monkey cube cost TENFOLD? What does that even have to do with food?

MTandi commented:
> Why does this increase monkey cube cost TENFOLD? What does that even have to do with food?

Ir was an unintended nerf by @ArcaneMusic in the materials update PR. The point is that monkey provides a lot of meat that can be turned into a lot of biomass. And you could get biomass out of nowhere.

Panzerkampfwaggen commented:
I believe this PR breaks golems. They can't eat anything since they can only eat minerals, tried using both raw and processed minerals they just won't consume it

san7890 commented:
conflicts
